### PR TITLE
mail js ref

### PR DESCRIPTION
**Pull request by alexkuhn**
Linked to https://github.com/odoo/enterprise/pull/4427

Task-ID 1914207

### CODE REVIEW

File: addons/mail/static/src/xml/wip/widget/systray_messaging_menu_dropdown.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        </div>
+        <div t-else="1" class="o_mail_systray_dropdown_top">
+            <div>
+                <button t-on-click='_onClickFilter' type="button" t-attf-class="btn btn-link o_filter_button {{!state.filter ? 'active' : ''}}" role="tab"> All </button>
+                <button t-on-click='_onClickFilter' type="button" t-attf-class="btn btn-link o_filter_button {{state.filter === 'chat' ? 'active' : ''}}" data-filter='chat' role="tab"> Chat </button>
+                <button t-on-click='_onClickFilter' type="button" t-attf-class="btn btn-link o_filter_button {{state.filter === 'channels' ? 'active' : ''}}" data-filter='channels' role="tab"> Channels </button>
+            </div>
+            <button type="button" class="btn btn-link o_new_message"> New message </button>
+        </div>
+        <t t-widget="ThreadPreviewList" class="o_mail_systray_dropdown_items" t-props="{ filter: state.filter }"/>
```
Comment (mge-odoo) [inline addons/mail/static/src/xml/wip/widget/systray_messaging_menu_dropdown.xml]: should not re-render it each time a message arrives if the dropdown is not opened?

**Comment by mge-odoo** [comment]
Also note that existing tests should be used and converted, no rewritten.

File: addons/mail/static/src/xml/owl/component/attachment.xml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+                    class="
+                        o_Attachment_asideItem
+                        o_Attachment_asideItemDownload"
+                    title="Download"
+                    t-if="
+                        props.isDownloadable and
+                        !storeProps.attachment.isTemporary"
+                    t-key="'download'"
+                    t-on-click="_onClickDownload">
+                    <i
```
Comment (seb-odoo) [inline addons/mail/static/src/xml/owl/component/attachment.xml]: generally, I think this is way too many lines for just 4 nested divs with images

File: addons/mail/static/tests/chatter_tests.js [hunk 3]
```diff
@@ -187,11 +187,11 @@ QUnit.module('Chatter', {
                 },
                 records:[
                     {id:1, type:'url', mimetype:'image/png', name:'filename.jpg',
-                     res_id: 7, res_model: 'partner'},
+                        res_id: 7, res_model: 'partner'},
```
Comment (seb-odoo) [inline addons/mail/static/tests/chatter_tests.js]: useless diff

File: addons/mail/static/src/js/owl/store/messaging_state.js [hunk 4]
```diff
@@ -0,0 +1,240 @@
+odoo.define('mail.store.state', function (require) {
+"use strict";
```
Comment (seb-odoo) [inline addons/mail/static/src/js/owl/store/messaging_state.js]: single quote?

File: addons/mail/static/src/js/owl/store/messaging_state.js [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+        threads: {},
+        threadCaches: {},
+    };
+    if (alteration) {
+        state = Object.assign(state, alteration);
+    }
+    return state;
+}
+
+return { init };
```
Comment (seb-odoo) [inline addons/mail/static/src/js/owl/store/messaging_state.js]: object wrapper seems useless?

File: addons/mail/static/src/js/owl/service/store_service.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+const EnvMixin = require('mail.widget.EnvMixin');
+
+const AbstractService = require('web.AbstractService');
+const config = require('web.config');
+const core = require('web.core');
+
+const DEBUG = true;
+
+const StoreService = AbstractService.extend(EnvMixin, {
+    TEST_ENV: {
```
Comment (seb-odoo) [inline addons/mail/static/src/js/owl/service/store_service.js]: strange to put test code in business code

File: addons/mail/static/src/js/owl/store/messaging_actions.js [hunk 7]
```diff
@@ -0,0 +1,3755 @@
...
+    async renameThread({ env, state }, threadLocalId, newName) {
+        const thread = state.threads[threadLocalId];
+        if (thread.channel_type === 'chat') {
+            await env.rpc({
+                model: 'mail.channel',
+                method: 'channel_set_custom_name',
+                args: [thread.id],
+                kwargs: {
+                    name: newName,
+                }
```
Comment (seb-odoo) [inline addons/mail/static/src/js/owl/store/messaging_actions.js]: missing ,

File: addons/mail/static/src/js/owl/store/messaging_actions.js [hunk 8]
```diff
@@ -0,0 +1,3755 @@
...
+     * @param {Object} param0.state
+     * @param {string} threadLocalId
+     * @param {string} newName
+     */
+    async renameThread({ env, state }, threadLocalId, newName) {
+        const thread = state.threads[threadLocalId];
+        if (thread.channel_type === 'chat') {
+            await env.rpc({
+                model: 'mail.channel',
+                method: 'channel_set_custom_name',
```
Comment (seb-odoo) [inline addons/mail/static/src/js/owl/store/messaging_actions.js]: try to remove this method and use `mail.channel.partner` write directly

**Pull request closed without merging**